internal/app/init/pkg/system/system.go

```diff
@@ -42,64 +47,57 @@ type Service interface {
 // nolint: golint
 func Services(data *userdata.UserData) *singleton {
 	once.Do(func() {
-		instance = &singleton{UserData: data}
+		instance = &singleton{
+			UserData: data,
+			State:    make(map[string]*ServiceRunner),
+		}
 	})
 	return instance
 }
 
-func runService(runnr runner.Runner) error {
-	if runnr == nil {
-		// special case - run nothing (TODO: we should handle it better, e.g. in PreFunc)
-		return nil
+// Start will invoke the service's Pre, Condition, and Type funcs. If the any
+// error occurs in the Pre or Condition invocations, it is up to the caller to
+// to restart the service.
+func (s *singleton) Start(services ...Service) {
+	s.mu.Lock()
+	if s.terminating {
```

Comment:
Should we add any sort of backoff delay on this?

Reply:
I'm not sure I follow this one, I was trying to make sure we never have a race for `Start` after `Shtudown` was called. This might happen happen, as some services are started from goroutine, and init's main goroutine might be going through the shutdown sequence already. 

If we do `wg.Add()` concurrently with `wg.Wait()`, this will explode iirc